src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -961,25 +961,79 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
     return matched
 
 
-def create_function(resource_group_name, new_app_name, storage_account, plan=None):
+def create_function(resource_group_name, name, storage_account, plan=None,
+                    consumption_plan_location=None):
+
+    if plan is not None and consumption_plan_location is not None:
+        raise CLIError("Use Plan or Consumption Plan Location")
```

Comment:
Think a validator would be the appropriate place for this logic. Uses would see a better error message.